inventory/linux.cf

```diff
@@ -7,6 +7,19 @@ bundle common inventory_linux
 #  systemd class based on linktarget of /proc/1/cmdline
 {
   vars:
+    have_proc_device_tree::
+      "proc_device_tree_model" string => readfile("/proc/device-tree/model"),
+      comment => "Read model from /proc/device-tree",
+      meta => { "inventory", "attribute_name=System version" };
+
+      "proc_device_tree_serial_number" string => readfile("/proc/device-tree/serial-number"),
+      meta => { "inventory", "attribute_name=System serial number" };
+
+      # the compatible proc file here separates groups of entries with a null byte, so replace that with a comma and get several descriptive classes
+      # e.g. raspberrypi,4-model-b\000brcm,bcm2711\000
+      "proc_device_tree_compatible" string => execresult("$(paths.cat) /proc/device-tree/compatible | $(paths.sed) 's/\x00/,/'", useshell);
+      "proc_device_tree_compatible_list" slist => splitstring("$(proc_device_tree_compatible)", ",", 20);
```

Comment:
Were you trying to get `raspberrypi_4_model_b` or `raspberrypi` and `4_model_b` classes a bit later https://github.com/cfengine/masterfiles/pull/2438/files#diff-3ee6fca00d4a49c79bc30911ddb39159e79c4d14241fc2aeb38dc8de0076eef8R74

Reply:
if you want `raspberrypi_4_model_b` then maybe replace the null byte with something else to split on instead of comma, like `MYSPLIT`

Reply:
I thought each individual thing made more sense to me: `raspberrypi` as a class seems VERY useful and of the two devices I sampled all of the individual bits seemed appropriate:

`raspberrypi, 4-model-b, brcm, bcm2711`

`pine64, pinephone-1.2, pinephone, allwinner, sun50i-a64`

It's fairly arbitrary but I thought it would be a good place to start.

Reply:
Defining `raspberrypi`, `raspberrypi_4`, and `raspberrypi_4_model_b` make a lot of sense to me. The `4_model_b` without `raspberrypi` in front - not so much.

Reply:
agreed, was just trying a catch all without much fiddling.

Reply:
see https://tracker.mender.io/browse/ENT-9029, let's defer parsing this business for now. :)